src/metabase/driver/postgres/actions.clj

```diff
@@ -27,38 +31,55 @@
-            :table      table
-            :ref-table  ref-table
-            :constraint constraint
-            :column     column})
-         columns)))))
+(defmethod sql-jdbc.actions/maybe-parse-sql-error [:postgres actions.error/violate-foreign-key-constraint]
+  [_driver error-type database error-message]
+  (when-let [[_match _table constraint ref-table _columns _value]
+             (re-find #"ERROR:\s+update or delete on table \"([^\"]+)\" violates foreign key constraint \"([^\"]+)\" on table \"([^\"]+)\"" error-message)]
+    (let [columns (constraint->column-names database constraint)]
+      {:type    error-type
+       :message (tru "Column(s) {0} is referenced from {1} table" (str/join ", " columns) ref-table)
+       :errors  (reduce (fn [acc col]
+                          (assoc acc col (tru "The value is referenced from {0} table" ref-table)))
+                        {}
+                        columns)})))
 
-(defmethod sql-jdbc.actions/parse-sql-error :postgres
-  [_driver database message]
-  (some #(% database message)
-        [maybe-parse-not-null-error
-         maybe-parse-unique-constraint-error
-         maybe-parse-fk-constraint-error]))
+(defmethod sql-jdbc.actions/maybe-parse-sql-error [:postgres actions.error/incorrect-value-type]
+  [_driver error-type _database error-message]
+  (when-let [[_ expected-type value]
+             (re-find #"ERROR:\s+invalid input syntax for type ([^\"]+):\s+\"([^\"]+)\"" error-message)]
+    (def incorrect-value-msg error-message)
+    {:type          error-type
+     :message (tru "Invalid value \"{0}\", expect type: {1}" value expected-type)
+     :errors  {}}))
+
+(comment
+ (sql-jdbc.actions/maybe-parse-sql-error
+  :postgres actions.error/violate-foreign-key-constraint {:id 47}
+  "ERROR: update or delete on table \"group\" violates foreign key constraint \"user_group-id_group_-159406530\" on table \"user\"\n  Detail: Key (id)=(1) is still referenced from table \"user\".")
+
+ (sql-jdbc.actions/maybe-parse-sql-error
+  :postgres actions.error/violate-unique-constraint {:id 47}
+  "Batch entry 0 UPDATE \"public\".\"group\" SET \"ranking\" = CAST(2 AS INTEGER) WHERE \"public\".\"group\".\"id\" = 1 was aborted: ERROR: duplicate key value violates unique constraint \"group_ranking_key\"\n  Detail: Key (ranking)=(2) already exists.  Call getNextException to see other errors in the batch.")
+
+ (sql-jdbc.actions/maybe-parse-sql-error
+  :postgres actions.error/violate-not-null-constraint nil
+  "ERROR: null value in column \"ranking\" of relation \"group\" violates not-null constraint\n  Detail: Failing row contains (57, admin, null).")
+
+ (sql-jdbc.actions/maybe-parse-sql-error
+  :postgres actions.error/incorrect-value-type nil
+  "Batch entry 0 UPDATE \"public\".\"group\" SET \"ranking\" = CAST('S' AS INTEGER) WHERE \"public\".\"group\".\"id\" = 1 was aborted: ERROR: invalid input syntax for type integer: \"S\"  Call getNextException to see other errors in the batch."))
```

Comment:
I'll add more tests later, but I don't think we should have error strings as test cases, I want to go further and construct the actually query to throw such errors so that if message error changes between driver version, we can catch that and write more robust regex to cover those cases.